src/drjt/java/com/lambdaherding/edi/drjt/ch04/Ex4.java

```diff
@@ -0,0 +1,34 @@
+package com.lambdaherding.edi.drjt.ch04;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.lambdaherding.edi.ch04.Artist;
+
+public class Ex4 {
+
+	// Copy-pasted from Artists
+	private List<Artist> artists;
+
+	public Ex4( List<Artist> artists ) {
+		this.artists = artists;
+	}
+
+//	public Artist getArtist( int index ) {
+//		if ( index < 0 || index >= artists.size() ) {
+//			indexException( index );
+//		}
+//		return artists.get( index );
+//	}
+
+	public String getArtistName( int index ) {
+		return getArtist( index ).map( Artist::getName ).orElse( "unknown" );
+	}
+	
+	public Optional<Artist> getArtist( int index ) {
+      if (index < 0 || index >= artists.size()) {
```

Comment:
There are ways to rewrite this in a more Java-8-ey way (i.e. avoiding the conditional), but it's a matter of debate what's clearest.